### PR TITLE
improvement(planner): Minor refactor of binder

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Refine code structure of binder.

## Changelog

- Improvement

### CODE REVIEW

**Review by andylokandy — APPROVED** [review]

**Comment by BohuTANG** [comment]
There are two CI failures:

Developing / test_stateless_standalone_linux (pull_request) Failing after 4m — test_stateless_standalone_linux
Developing / test_stateless_cluster_linux (pull_request) Failing after 6m — test_stateless_cluster_linux

https://github.com/datafuselabs/databend/runs/6415239350?check_suite_focus=true#step:3:502

**Comment by BohuTANG** [comment]
@mergify update

File: query/src/sql/planner/binder/sort.rs [hunk 1]
```diff
@@ -13,42 +13,58 @@
...
+        for order in order_by {
+            // First we try to resolve sort item with `SELECT` context
+            let mut res = select_scalar_binder.bind_expr(&order.expr).await;
+            // If failed, we will try to resolve sort item with `FROM` context
+            if let Err(e) = res.clone() {
+                // If still failed, the previous error will be raised
+                res = from_scalar_binder.bind_expr(&order.expr).await.or(Err(e));
+            }
+            let (scalar, _) = res?;
+            if !matches!(scalar, Scalar::BoundColumnRef(_)) {
```
Comment (xudong963) [inline query/src/sql/planner/binder/sort.rs:48]: Similar statements such as `select a from t order by a + b` don't seem to be allowed here?

Comment (leiysky) [reply query/src/sql/planner/binder/sort.rs:48]: It's unclear. We will only support order by columns can be displayed in result set.

**Pull request merged**